File: Roadmap/01 - OPERADORES Y ESTRUCTURAS DE CONTROL/javascript/Roadmap/00 - SINTAXIS, VARIABLES, TIPOS DE DATOS Y HOLA MUNDO/rust/darubiano.rs
/*
    EJERCICIO:
    1) Crea un comentario en el código y coloca la URL del sitio web oficial del lenguaje de programación que has seleccionado.
    2) Representa las diferentes sintaxis que existen de crear comentarios en el lenguaje (en una línea, varias...).
    3) Crea una variable (y una constante si el lenguaje lo soporta).
    4) Crea variables representando todos los tipos de datos primitivos del lenguaje (cadenas de texto, enteros, booleanos...).
    5) Imprime por terminal el texto: "¡Hola, [y el nombre de tu lenguaje]!"
*/

// 1) https://www.rust-lang.org/es/

// 2) Comentario de una linea y varias

/*
Comentario de varias lineas
*/
fn main() {
    // 3) Variables son (inmutables en Rust por defecto) y constante
    let variable: &str = "variable";
    let mut variable_mutable: &str = "variable_mutable";
    // Constantes
    const MY_CONST: &str = "constante";

    // 4) Tipos de variables primitivas https://doc.rust-lang.org/rust-by-example/primitives.html
    //Tipos numericos
    // Enteros con signo
    let int: i8 = 1;
    let int_suffix = 1i8;
    let int_i16: i16 = 1;
    let int_i32: i32 = 1;
    let int_i64: i64 = 1;
    let int_i128: i128 = 1;
    // Enteros sin signo
    let uint_u16: u16 = 1;
    let uint_u32: u32 = 1;
    let uint_u64: u64 = 1;
    let uint_u128: u128 = 1;
    // Coma flotante
    let float_f32: f32 = 1.0;
    let float_f64: f64 = 1.0;

    // Tipo de dato texto
    let caracter: char = 'a';
    let variable: &str = "variable";
    let mut texto_mutable: String = String::from("¡Hola, ");

    // Tipos boleanos
    let verdadero: bool = true;
    let falso: bool = false;

    // Tipos arreglos,slices, tuplas y unidad
    let array: [i32; 3] = [1, 2, 3];
    let slice: &[i32] = &array[1..3];
    let tupla: (i32, f64, char) = (42, 3.14, 'A');
    let unidad: () = ();

    // 5) print() hola Go
    println!("¡Hola, Rust!");
}

File: Roadmap/00 - SINTAXIS, VARIABLES, TIPOS DE DATOS Y HOLA MUNDO/go/darubiano.go
/*
   EJERCICIO:
   1) Crea un comentario en el código y coloca la URL del sitio web oficial del lenguaje de programación que has seleccionado.
   2) Representa las diferentes sintaxis que existen de crear comentarios en el lenguaje (en una línea, varias...).
   3) Crea una variable (y una constante si el lenguaje lo soporta).
   4) Crea variables representando todos los tipos de datos primitivos del lenguaje (cadenas de texto, enteros, booleanos...).
   5) Imprime por terminal el texto: "¡Hola, [y el nombre de tu lenguaje]!"
*/

// 1) https://go.dev/doc/

// 2) Comentario de una linea y varias

/*
Comentario de varias lineas
*/
package main

import "fmt"

func main() {
	// 3) Variable y constante
	const pi float64 = 3.1416
	var variable string = "Go"
	otra_variable := "otra manera de declarar variable"

	// 4) Tipos de variables primitivas https://go.dev/learn/
	//Tipos numericos
	// unsigned integers
	var _uint uint = 18446744073709551615
	var _uint8 uint8 = 255
	var _uint16 uint16 = 65535
	var _uint32 uint32 = 4294967295
	var _uint64 uint64 = 18446744073709551615
	// signed integers
	var _int int = -2147483648
	var _int8 int8 = 127
	var _int16 int16 = 32767
	var _int32 int32 = 2147483647
	var _int64 int64 = 9223372036854775807
	// float point numbers
	var _float32 float32 = 1234.5
	var _float64 float64 = 13412341212342.3434342432
	// complex numbers
	var _complex64 complex64 = 1i
	var _complex128 complex128 = 1e64i

	// Tipos boleanos
	var _true bool = true
	var _false bool = false

	//Tipos lista o slice
	var _slice_string []string = []string{"1", "2", "3"}
	var _slice_int []int = []int{1, 2, 3}

	// Tipo de dato
	var _map_string_string map[string]string = map[string]string{"1": "first", "2": "second", "3": "third"}
	var _map_string_int map[string]int = map[string]int{"first": 1, "second": 2, "third": 3}

	// 5) print() hola Go
	fmt.Println("¡Hola, Go!")
	fmt.Println(
		pi,
		variable,
		otra_variable,
		_uint,
		_uint8,
		_uint16,
		_uint32,
		_uint64,
		_int,
		_int8,
		_int16,
		_int32,
		_int64,
		_float32,
		_float64,
		_complex64,
		_complex128,
		_true,
		_false,
		_slice_string,
		_slice_int,
		_map_string_string,
		_map_string_int)
}
